aws/sagemaker/llm-instance-recommend/lmi_recommender/lmi_config_recommender.py

```diff
@@ -111,8 +126,13 @@ def category_checker(target, category):
     return found
 
 
-def rolling_batch_chooser(model_category: str, instance: dict, dtype: str):
+def strategy_chooser(model_category: str, instance: dict, dtype: str):
     instance_name = instance['name']
+    smoothquant_checker = dynamic_batch_deepspeed['smoothquant']
+    if dtype == 'int8':
```

Comment:
Where is this `dtype` coming from? I think `dtype` should actually be chosen independently of quantization type. This is how smoothquant config works.

You never quantize all operations in a model, and the non-quantized parts may be better served as bf16 rather than fp16 for some models (if they were trained in bf16). Our smoothquant implementation supports using dtype fp16 or bf16.

Reply:
I think "dtype int8" is the wrong API to use for quantization. Not every layer is quantized, so the model still needs to run non-quantized layers with some dtype.

But, it's not that big of a deal. float16 is the right default, generally speaking, for now (which is true for non-quantized inference, too).